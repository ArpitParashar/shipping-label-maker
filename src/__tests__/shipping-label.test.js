import React from 'react';
import ShippingLabel from '../features/shipping-label-maker/shipping-label';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import Stepper from '@material-ui/core/Stepper';

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

describe('ShippingLabel component', () => {
  it('should render using context', () => {
    useContextMock.mockReturnValue([{
        from: {
          name: '',
          street: '',
          city: '',
          state: '',
          zip: '',
        },
        to: {
          name: '',
          street: '',
          city: '',
          state: '',
          zip: '',
        },
        weight: '',
        shippingOption: '',
      }, ()=>{}]);

    const element = new ShallowRenderer().render(
      <ShippingLabel />
    );
    expect(element.props.children).toMatchSnapshot();
  });

  it('should calculate correct shipping based on weight and mode of shipping', () => {
    useContextMock.mockReturnValue([{
        from: {
          name: '',
          street: '',
          city: '',
          state: '',
          zip: '',
        },
        to: {
          name: '',
          street: '',
          city: '',
          state: '',
          zip: '',
        },
        weight: 12,
        shippingOption: 1,
      }, ()=>{}]);
   
    const element = shallow(<ShippingLabel />);
    expect(element.find('#shippingcCost').filterWhere((item) => {
        return item.prop('secondary') === '4.80';
      })).toHaveLength(1);
  });
});
