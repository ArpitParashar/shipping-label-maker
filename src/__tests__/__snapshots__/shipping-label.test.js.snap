exports[`ShippingLabel component should render using context 1`] = `
Array [
  <h3
    className="red-text center text-darken-3">
     Your shipping label!
</h3>,
  <Unknown
    className="container">
    <Unknown>
        <div
            className="row">
            <div
                className="col s6">
                <h6>
                    FROM:
                </h6>
                <Unknown>
                    <Unknown>
                        <Unknown
                            primary="Name"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="Street"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="City"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="State"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="Zip"
                            secondary="" />
                    </Unknown>
                </Unknown>
            </div>
            <div
                className="col s6">
                <h6>
                    TO:
                </h6>
                <Unknown>
                    <Unknown>
                        <Unknown
                            primary="Name"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="Street"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="City"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="State"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="Zip"
                            secondary="" />
                    </Unknown>
                </Unknown>
            </div>
        </div>
        <div
            className="row">
            <div
                className="col s12">
                <Unknown>
                    <Unknown>
                        <Unknown
                            primary="Weight"
                            secondary="" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            primary="Shipping Option"
                            secondary="Priority" />
                    </Unknown>
                    <Unknown>
                        <Unknown
                            id="shippingcCost"
                            primary="Shipping Cost"
                            secondary="0.00" />
                    </Unknown>
                </Unknown>
            </div>
        </div>
    </Unknown>
    <Unknown
        className="right">
        <Unknown
            color="secondary"
            onClick={[Function]}
            variant="contained">
            Create Another!
        </Unknown>
    </Unknown>
</Unknown>,
]
`;
